app/src/test/java/org/schabi/newpipe/database/playlist/PlaylistLocalItemTest.java

```diff
@@ -0,0 +1,105 @@
+package org.schabi.newpipe.database.playlist;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.schabi.newpipe.database.playlist.model.PlaylistRemoteEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PlaylistLocalItemTest {
+    @Test
+    public void emptyPlaylists() {
+        final List<PlaylistMetadataEntry> localPlaylists = new ArrayList<>();
+        final List<PlaylistRemoteEntity> remotePlaylists = new ArrayList<>();
+        final List<PlaylistLocalItem> mergedPlaylists =
+                PlaylistLocalItem.merge(localPlaylists, remotePlaylists);
+
+        assertEquals(0, mergedPlaylists.size());
+    }
+
+    @Test
+    public void onlyLocalPlaylists() {
+        final List<PlaylistMetadataEntry> localPlaylists = new ArrayList<>();
+        final List<PlaylistRemoteEntity> remotePlaylists = new ArrayList<>();
+        localPlaylists.add(new PlaylistMetadataEntry(1, "name1", "", 0, 1));
+        localPlaylists.add(new PlaylistMetadataEntry(2, "name2", "", 1, 1));
+        localPlaylists.add(new PlaylistMetadataEntry(3, "name3", "", 3, 1));
```

Comment:
Shouldn't the indexes be in a random order, so that the test can check whether `merge` sorts correctly? Also in other functions here

Reply:
See https://github.com/TeamNewPipe/NewPipe/pull/8221#discussion_r1545370806